src/hip_memory.cpp

```diff
@@ -1509,6 +1509,13 @@ __global__ void hip_copy2d_n(T* dst, const T* src, size_t width, size_t height,
 
 template <typename T>
 void ihipMemsetKernel(hipStream_t stream, T* ptr, T val, size_t sizeBytes) {
+
+    if(sizeof(T) == sizeof(uint32_t)){
+        // for all sizeof(uint32_t) use the fast path using hsa specialization
+        hsa_amd_memory_fill(ptr, reinterpret_cast<const std::uint32_t&>(val), sizeBytes);
```

Comment:
Failure is in one of the memset tests. I need to check the hsa spec, but I think the ```sizeBytes``` probably needs to be divided by ```sizeof(uint32_t)```.

Reply:
According to https://github.com/RadeonOpenCompute/ROCR-Runtime/blob/d823418d0500b30af775efb3b17b28e164958f3a/src/inc/hsa_ext_amd.h#L1282 you are right! Fixing.

Reply:
@mangupta looks like that's not sufficient to pass tests - mind having a look? Thanks!